### PR TITLE
[runtime-security] add kernel lockdown sanity check

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR add some sanity checks at runtime module start. It checks debugfs and kernel lockdown.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]

**Comment by safchain** [comment]
@brycekahle thanks for the review. I changed a bit the PR, more specifically I enforced the kprobe_events check. It would nice if could have another look.

File: pkg/util/kernel/fs.go [hunk 1]
```diff
@@ -3,23 +3,83 @@
...
 			return false, fmt.Errorf("debugfs is not mounted and is needed for eBPF-based checks, run \"sudo mount -t debugfs none /sys/kernel/debug\" to mount debugfs")
 		} else {
 			return false, fmt.Errorf("an error occurred while accessing debugfs: %w", err)
 		}
 	}
+
+	mi, err := GetMountPoint("/sys/kernel/debug/tracing/kprobe_events")
+	if err != nil {
+		return false, errors.New("unable to detect debugfs mount point")
+
```
Comment (brycekahle) [inline pkg/util/kernel/fs.go:70]: nit: extra empty line here

File: pkg/util/kernel/fs.go [hunk 2]
```diff
@@ -3,23 +3,83 @@
...
+	}
+
+	if mi.FSType != "tracefs" {
```
Comment (brycekahle) [inline pkg/util/kernel/fs.go:73]: On my kernel, the mountpoint that would match for `/sys/kernel/debug/tracing/kprobe_events` is:
`53 24 0:8 / /sys/kernel/debug rw,nosuid,nodev,noexec,relatime shared:29 - debugfs debugfs rw`
and thus the type would be `debugfs`.

Do we need to check for `debugfs` as well?

Comment (safchain) [reply pkg/util/kernel/fs.go:73]: I'll add `debugfs`, on my side `tracefs` is auto-mounted.

Comment (brycekahle) [reply pkg/util/kernel/fs.go:73]: I see `tracefs` too, but under `/sys/kernel/tracing`

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]

File: pkg/util/kernel/fs.go [hunk 3]
```diff
@@ -3,23 +3,82 @@
...
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/moby/sys/mountinfo"
 )
 
+// MountInfoPidPath returns the path to the mountinfo file of a pid in /proc
+func MountInfoPidPath(pid int32) string {
+	return filepath.Join(util.HostProc(), fmt.Sprintf("/%d/mountinfo", pid))
```
Comment (brycekahle) [inline pkg/util/kernel/fs.go:18]: Couldn't this be wrong if the host proc is mapped at `/host/proc`? If the container is in a different PID namespace, the PID returned by `os.Getpid()` will not be correct in `/host/proc`, only correct in `/proc`?

**Pull request merged**